x-pack/plugins/threat_intelligence/public/modules/indicators/components/indicator_value_actions/indicator_value_actions.tsx

```diff
@@ -54,9 +54,9 @@ export const IndicatorValueActions: VFC<IndicatorValueActions> = ({
 
   return (
     <EuiFlexGroup justifyContent="center" alignItems="center">
-      <FilterIn as={Component} data={indicator} field={field} data-test-subj={filterInTestId} />
-      <FilterOut as={Component} data={indicator} field={field} data-test-subj={filterOutTestId} />
-      <AddToTimeline
+      <FilterInButtonIcon data={indicator} field={field} data-test-subj={filterInTestId} />
+      <FilterOutButtonIcon data={indicator} field={field} data-test-subj={filterOutTestId} />
+      <AddToTimelineButtonIcon
         as={Component}
```

Comment:
what is this `as={Component}` mean/do? I couldn't trace it back through all the prop spreading but I tried quite hard :)

Reply:
it tells the AddToTimelineButtonIcon what component should be used for the display, but then the naming on the `AddToTimelineButtonIcon` is a bit misleading. Maybe `as` which is a generic convention I saw in a couple of libraries is not right here :D maybe, if it is about changing the icon, the prop should be called `icon`.

Reply:
yeah looking at the EuiDataGrid doc, this is how they describe the `Component` props of the `EuiDataGridColumnCellActionProps` interface:

_React component representing the action displayed in the cell_

Reply:
looking at the `AddToTimelineButtonProps` doc, the `Component` props says

_`Component` is only used with `EuiDataGrid`; the grid keeps a reference to `Component` for show / hide functionality_

Reply:
I originally had the props named `Component`. I can either go back to that to be consistent with what's used in Eui library, or use `icon` or something else...

Reply:
yeah, now I realize what is confusing for me:
1. When the component is further away from the Data Grid and Cell Action it's very hard to make this connection, especially if it named like that. What about explicitly calling the prop `cellActionComponent` instead of `as`?
2. `ButtonIcon` part in the name which doesn't correlate explicitly with the implementation of the component. The whole button/icon button thing happens implicitly through the `Component` passed via cellActions and this connection is lost when I'm down in the component tree. does it make sense to have a more generic name `addToTimelineAction` for instance?
3. the data grid docs say https://elastic.github.io/eui/#/tabular-content/data-grid-cells-popovers#cell-actions `Because different button types are used between the cell and the cell popover, we pass your render function a Component prop which you must render in order for your cell actions to switch correctly between button icons and buttons.` . So Component can be either Button or Icon Button somehow automagically depending on the logic inside the data grid. Do we do smth specific so it is only being rendered as and Icon Button?

Reply:
> ButtonIcon part in the name which doesn't correlate explicitly with the implementation of the component. The whole button/icon button thing happens implicitly through the Component passed via cellActions and this connection is lost when I'm down in the component tree. does it make sense to have a more generic name addToTimelineAction for instance?

I just actually thought that I could be more verbose and - similary to the FilterIn and FilterOut components - I could separate this into 2 components, one for the DataGrid which would have this Component props, and the other that would be a simple ButtonIcon that wouldn't have the props.
That was we're explicitly clear on when to use the components. Extracting the rest of the logic in a custom hook would avoid too much duplication...

> the data grid docs say https://elastic.github.io/eui/#/tabular-content/data-grid-cells-popovers#cell-actions Because different button types are used between the cell and the cell popover, we pass your render function a Component prop which you must render in order for your cell actions to switch correctly between button icons and buttons. . So Component can be either Button or Icon Button somehow automagically depending on the logic inside the data grid. Do we do smth specific so it is only being rendered as and Icon Button?

We don't do anything specific, the magic happens inside the EuiDataGrid component...